eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml

```diff
@@ -338,4 +338,30 @@
     <Field name="~(MAX_DURATION|NULL_TXN)"/>
     <Bug pattern="MS_SHOULD_BE_FINAL"/>
   </Match>
+
+  <!--  The return value is ignored for purpose-->
+  <Match>
+    <Class name="class com.microsoft.azure.servicebus.primitives.AsyncUtil"/>
+    <Method name="~(completeFuture|completeFutureExceptionally)"/>
+    <Bug pattern="RV_RETURN_VALUE_IGNORED_BAD_PRACTICE"/>
+  </Match>
+
+  <!--  The casting is correct, it is beyond the SpotBugs's ability to determine the correctness -->
+  <Match>
+    <Class name="com.microsoft.azure.servicebus.primitives.RequestResponseUtils"/>
+    <Method name="encodeRuleDescriptionToMap"/>
+    <Bug pattern="BC_UNCONFIRMED_CAST_OF_RETURN_VALUE"/>
+  </Match>
+
+  <!--  Bytes array are designed to return null if it is null-->
+  <Match>
+    <Class name="com.microsoft.azure.servicebus.Utils"/>
+    <Method name="getDataFromMessageBody"/>
+    <Bug pattern="PZLA_PREFER_ZERO_LENGTH_ARRAYS"/>
+  </Match>
+
+  <Match>
+    <Class name="com.microsoft.azure.servicebus.primitives.MessageWithDeliveryTag"/>
```

Comment:
[SpotBugs-P2]

This is P2 SpotBugs suppression.

com.microsoft.azure.servicebus.primitives.MessageWithDeliveryTag.getDeliveryTag() may expose internal representation by returning MessageWithDeliveryTag.deliveryTag. 

Returning a reference to a mutable object value stored in one of the object's fields exposes the internal representation of the object.  If instances are accessed by untrusted code, and unchecked changes to the mutable object would compromise security or other important properties, you will need to do something different. Returning a new copy of the object is better approach in many situations.



new com.microsoft.azure.servicebus.primitives.MessageWithDeliveryTag(Message, byte[]) may expose internal representation by storing an externally mutable object into MessageWithDeliveryTag.deliveryTag.

This code stores a reference to an externally mutable object into the internal representation of the object.  If instances are accessed by untrusted code, and unchecked changes to the mutable object would compromise security or other important properties, you will need to do something different. Storing a copy of the object is better approach in many situations.

Reply:
Is there a reason why these are suppressed rather than cloning the byte[] array? You did it in a couple of PRs back.